components/Feed/StickySection.js

```diff
@@ -0,0 +1,127 @@
+import { Component } from 'react'
+import { HEADER_HEIGHT, HEADER_HEIGHT_MOBILE } from '../constants'
+import { css } from 'glamor'
+import { mediaQueries, colors } from '@project-r/styleguide'
+import PropTypes from 'prop-types'
+
+const SIDEBAR_WIDTH = 120
+const MARGIN_WIDTH = 20
+const STICKY_HEADER_HEIGHT = 27
+
+const style = {
+  header: css({
+    backgroundColor: '#fff',
+    margin: '0 0 30px 0',
+    width: '100%',
+    height: STICKY_HEADER_HEIGHT,
+    [mediaQueries.lUp]: {
+      height: 'auto',
+      float: 'left',
+      margin: `0 0 30px -${SIDEBAR_WIDTH + MARGIN_WIDTH}px`,
+      width: SIDEBAR_WIDTH,
+      '& > div': {
+        width: SIDEBAR_WIDTH
+      }
+    }
+  }),
+  label: css({
+    padding: '8px 0',
+    borderTop: '1px solid #000',
+    backgroundColor: '#fff'
+  }),
+  sticky: css({
+    top: HEADER_HEIGHT_MOBILE - 1,
+    borderBottom: `0.5px solid ${colors.divider}`,
+    [mediaQueries.mUp]: {
+      top: HEADER_HEIGHT - 1
+    },
+    [mediaQueries.lUp]: {
+      borderBottom: 'none'
+    }
+  })
+}
+
+class StickySection extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      sticky: false,
+      isMobile: true,
+      isSmall: false,
+      width: 0,
+      height: 0
+    }
+    this.sectionRef = null
+    this.setSectionRef = (el) => { this.sectionRef = el }
+
+    this.onScroll = () => {
+      if (this.sectionRef) {
+        const { sticky, isSmall, height } = this.state
+        const { hasSpaceAfter } = this.props
+        const y = window.pageYOffset + (isSmall ? HEADER_HEIGHT_MOBILE : HEADER_HEIGHT)
+        const offset = this.sectionRef.offsetTop
+        const nextSticky = (y > offset) && // scroll pos is below top of section
+          (offset + height + (hasSpaceAfter ? STICKY_HEADER_HEIGHT : 0) > y) // scroll pos is above bottom
+        if (sticky !== nextSticky) {
+          this.setState({sticky: nextSticky})
+        }
+      }
+    }
+
+    this.measure = () => {
+      const isMobile = window.innerWidth < mediaQueries.lBreakPoint
+      const isSmall = window.innerWidth < mediaQueries.mBreakPoint
```

Comment:
Cool, I just wanted to comment on the jumping header in non-floating mode on desktop, but I've seen you just fixed it with introducing `isSmall`.

I just think the naming of `isMobile` could be misleading, since the header and the rest of the page are still in desktop mode.